File: compression-tool/typescript-huffman-compression/tests/compression.test.ts
import { count } from "console";
import { countFrequencies } from "../src/compression";

describe('countFrequencies', () => {
    it('should count the frequencies of characters correctly', () => {
        const text = 'aabbccabc';
        const expectedFrequencies = { a: 3, b: 3, c: 3 };
        expect(countFrequencies(text)).toEqual(expectedFrequencies);
    });
});

File: compression-tool/elixir-huffman-compression/huffman_compression/test/compression_test.exs
defmodule CompressionTest do 
  use ExUnit.Case 
  doctest Compression

  test "count_frequencies/1 counts the frequenciesof characters correctly" do 
    text = "aabbccabc"
    expected_frequencies = %{"a" => 3, "b" => 3, "c" => 3}
    assert Compression.count_frequencies(text) == expected_frequencies
  end
end
